internal/http/listener.go

```diff
@@ -136,20 +137,39 @@ func newHTTPListener(ctx context.Context, serverAddrs []string) (listener *httpL
 		}
 	}()
 
+	// Silently ignore failure to bind on DNS cached ipv6 loopback
 	for _, serverAddr := range serverAddrs {
 		var l net.Listener
 		if l, err = listenCfg.Listen(ctx, "tcp", serverAddr); err != nil {
+			if strings.HasPrefix(serverAddr, "[::1]") {
+				continue
+			}
 			return nil, err
 		}
 
 		tcpListener, ok := l.(*net.TCPListener)
 		if !ok {
-			return nil, fmt.Errorf("unexpected listener type found %v, expected net.TCPListener", l)
+			err = fmt.Errorf("unexpected listener type found %v, expected net.TCPListener", l)
+			if strings.HasPrefix(serverAddr, "[::1]") {
+				continue
+			}
+			return nil, err
 		}
 
 		tcpListeners = append(tcpListeners, tcpListener)
 	}
 
+	// Fail if no listeners found
+	if len(tcpListeners) == 0 {
+		// Report specific issue
+		if err != nil {
+			return nil, err
+		}
+		// Report general issue
+		err = fmt.Errorf("%v listeners found, expected at least 1", len(tcpListeners))
```

Comment:
since the if block already ensures the `tcpListeners` is `0 ` should we just update the error string to not use `len(tcpListeners))`?